---
title: "assignment1_2017250101_101_lee"
output: html_notebook
---

# 1. Loading the data

```{r}
d <- read.delim("DDG2P_24_9_2019.with_pLI.txt")
```

# 2. Draw the plot

We will draw two plots: 1) pLI score by allelic requirement and 2) pLI score by DDD category.
For this, we will use the ridgeline plot.

```{r}
library(ggridges)
library(ggplot2)

#Draw the first plot: pLI score by allelic requirement

p1 <- d %>% filter(allelic.requirement != "NA") %>%
  ggplot(aes(x = pLI, y = allelic.requirement, fill = allelic.requirement)) +
  geom_density_ridges(alpha=0.6, stat="binline", bins=50)+
  theme_ridges() + 
  theme(
    legend.position = "none",
    panel.spacing = unit(0.1, "lines")) +
  xlab("pLI score")+
  ylab("Allelic requirement")

p1
```


```{r}
#For this time, in order to change the figure color and also make the plot colorblind friendly, lets define the cbPalette and use it to fill the figure.

cbPalette <- c("#999999", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")

p2 <- ggplot(d, aes(x = pLI, y = DDD.category, fill = DDD.category)) +
  scale_fill_manual(values=cbPalette)+
  geom_density_ridges(scale=0.9, alpha=0.6, stat="binline", bins=50)+
  theme_ridges() + 
  theme(
    legend.position = "none",
    panel.spacing = unit(0.1, "lines")) +
  xlab("pLI score")+
  ylab("DDD category")
  

#Because the plots touches a line a lot above, we've changed a scale of the plot to 0.9. Scale, one of the aesthetics, is a scaling factor to scale the height of the ridgelines relative to the spacing between them. A value of 1 indicates that the maximum point of any ridgeline touches the baseline right above, assuming even spacing between baselines.

p2
```

# 3. Merge the plots

We will merge the previous two plots using the function `cowplot`.
```{r fig 1, fig.width=10, fig.height=5}
library(cowplot)
p<-plot_grid(p1,p2,ncol=2)
p
```

# 4. Explanation

The pLI score, which is computed from Exome Aggregation Consortium (ExAC), indicates the probability that a gene is intolerant to a Loss of Function (LoF) mutation. Genes with high pLI scores (pLI ≥ 0.9) are extremely LoF intolerant, whereby genes with low pLI scores (pLI ≤ 0.1) are LoF tolerant.

To begin with, if you see the plot in the left - pLI score by allelic requirement, you can see that the allelic requirement entries are divided into two groups: x-linked over-dominance, x-linked dominant, mosaic, monoallelic, imprinted, and hemizygous/mosaic, hemizygous have relatively high pLI score, which means that they are LoF intolerant. Therefore, the genes of these allelic requirement have a high probability to fall into the Haplosufficient category. 
However, digenic, biallelic/monoallelic, and biallelic have relatively low pLI score, which means that they are LoF tolerant. Therefore, the genes of these allelic requirement have a low probability to fall into the Haplosufficient category. 

We can understand this result related to the definition of Haplosufficient category. According to Medicinenet, the Haplosufficient category is defined as:

1. A situation in which the total level of a gene product (a particular protein) produced by the cell is about half of the normal level and that is not sufficient to permit the cell to function normally. 
2. A condition that arises when the normal phenotype requires the protein product of both alleles, and reduction of 50% of gene function results in an abnormal phenotype.

Simply, this situation is easily happened when the product of gene is insufficient, so hemizygous genes(including genes in sex chromosome) are intolerate to it. This is because there is no gene to replace the function if single allele is mutated.

Moreover, we can see the plot in the right - pLI score by DDD category. DDD is Deciphering Developmental Disorders, and the DDD category is the level of certainty that the gene causes developmental disease. As you see the plot, except 'child IF' category, the pLI scores of the categories are polarized from end to end. This states that even though it can easily be expected that the high pLI score would lead to high DDD probability because DDD is shown by the abnormal phenotype, the pLI score and the DDD probability do not have positive correlation. More accurately, as the pLI scores are highly polarized, there would be more than one other etiology that cause DDD, not only LoF.